crates/core/src/db/relational_db.rs

```diff
@@ -256,16 +272,29 @@ impl RelationalDB {
     /// state. See also [`Self::with_auto_commit`].
     #[tracing::instrument(skip_all)]
     pub fn begin_tx(&self) -> MutTx {
-        log::trace!("BEGIN TX");
+        log::trace!("BEGIN Mut TX");
         self.inner.begin_mut_tx()
     }
 
+    // TODO(shub): name this as per decided convention (mut_tx/ tx).     
+    #[tracing::instrument(skip_all)]
+    pub fn begin_read_tx(&self) -> Tx {
+        log::trace!("BEGIN TX");
+        self.inner.begin_tx()
+    }
+
     #[tracing::instrument(skip_all)]
     pub fn rollback_tx(&self, ctx: &ExecutionContext, tx: MutTx) {
-        log::trace!("ROLLBACK TX");
+        log::trace!("ROLLBACK Mut TX");
```

Comment:
```suggestion
        log::trace!("ROLLBACK MUT TX");
```

Just for consistency